viewer/packages/sector-loader/src/V8SectorRepository.ts

```diff
@@ -94,7 +94,9 @@ export class V8SectorRepository implements SectorRepository {
       }
     } catch (error) {
       this._consumedSectorCache.remove(cacheKey);
-      MetricsLogger.trackError(error as Error, { methodName: 'loadSector', moduleName: 'CachedRepository' });
+      if (error) {
+        MetricsLogger.trackError(error as Error, { methodName: 'loadSector', moduleName: 'CachedRepository' });
+      }
       throw error;
```

Comment:
I think this should be handled within MetricsLogger - even though its very odd that someone would do `throw undefined`, `MetricsLogger` should be more resilient to handling whatever it gets without barfing.

Reply:
I revisitted the code - it actually looks like `MetricsLogger` wasn't what caused the loading problem. So while this change doesn't really contribute to fixing the problem, it still fixes that `MetricsLogger` will print `undefined` in the console (as an error) if it is called with an undefined argument. And that message is very unhelpful to a user. 

I'll go forth in trying to find the source of the undefined error, as it seems like a better way to fix it.

Reply:
I found the source - it was at the bottom, with the low-level fetch calls. I have made sure they return better errors.